src/monty/json.py

```diff
@@ -21,47 +21,24 @@
 from typing import Any
 from uuid import UUID, uuid4
 
-try:
-    import numpy as np
-except ImportError:
-    np = None
-
-try:
-    import pydantic
-except ImportError:
-    pydantic = None
-
-try:
-    from pydantic_core import core_schema
-except ImportError:
-    core_schema = None
-
 try:
     import bson
 except ImportError:
     bson = None
 
-try:
-    from ruamel.yaml import YAML
-except ImportError:
-    YAML = None
-
 try:
     import orjson
 except ImportError:
     orjson = None
 
 
-try:
-    import torch
-except ImportError:
-    torch = None
-
 __version__ = "3.0.0"
 
 
 def _load_redirect(redirect_file):
     try:
+        from ruamel.yaml import YAML
```

Comment:
This PR doesn't change the overall workflow of `monty`, just a head up here, if `ruamel.yaml` is not install, trying to import `MSONable` would trigger `ImportError`.

Also note `monty` as a standalone package doesn't list any dependencies: https://github.com/materialsvirtuallab/monty/blob/1270c7b230b43cdbc4f5f184ac645ece31d49b61/pyproject.toml#L21-L23